scripts/dev/upstream_merge/upstream_merge.sh

```diff
@@ -0,0 +1,206 @@
+#!/bin/bash
+
+# This script can be used for merging latest upstream for each of the
+# repos described in repos.conf
+#
+# For each local repo, it does following
+# 1. Check if local_base_branch specified in repos.conf exists
+# 2. Checkout local_base_branch and run git pull
+# 3. Add a 'remote' named 'automerge_upstream'
+#    a. If the remote already exists, remove it first
+# 4. Pulls remote branch into a branch named 'dev/automerge/upstream/<upstream_branch>'
+#    a. If the branch name already exists, remove it first
+# 5. Creates a branch named 'dev/automerge/ni' from 'nilrt/master/hardknott' and checkout
+#    a. If the branch name exists, ask to delete or skip or cancel
+# 6. Merges 'dev/automerge/upstream' into 'dev/automerge/ni'
+#    a. Success or failure is reported; script continues in either case
+
+DIR=$(dirname ${BASH_SOURCE})
+CONF_FILE="$DIR/repos.conf"
+
+REMOTE_REPO_NAME="automerge_upstream"
+LOCAL_UPSTREAM_BRANCH_PREFIX="dev/automerge/upstream"
+LOCAL_BRANCH_NAME="dev/automerge/ni"
+
+usage() {
+   echo "Usage: $0 [-c <conf file>][-h] " 1>&2
+   exit 1
+}
+
+parse_args() {
+   local OPTIND o
+   while getopts "c:h" o; do
+      case "${o}" in
+         c)
+            CONF_FILE=${OPTARG}
+            ;;
+         h)
+            usage
+            ;;
+         *)
+            usage
+            ;;
+      esac
+   done
+   shift $((OPTIND-1))
+}
+
+sanity_test_repo() {
```

Comment:
Could you alphabetize the functions in this script so that they're easier to traverse?

Reply:
All the functions (except main) are currently laid out in the order in which they're called. I find that this is simpler to navigate as we can just read the file top down to understand code flow without digging inside the top level functions.